In [8]:
# Imports Library
import pandas as pd


# Utils Functions
def load_data(file: str):
    df = pd.read_csv(f"data/{file}.csv")
    return df

# CAR-SALES dataset

Dữ liệu `car-sales.csv`: bộ dữ liệu chứa thông tin các ôtô cũ ở một cửa hàng bán ô tô cũ. 

Bạn hãy thực hiện các yêu cầu sau:


### 1. Đọc dữ liệu và cho biết dữ liệu có bao nhiêu dòng, bao nhiêu cột

Đáp án: (10 dòng, 5 cột)

In [9]:
df = load_data("car-sales")

df.head()

Manufacturer Colour  Odometer  Doors    Price
0       Toyota  White    150043      4   4000.0
1        Honda    Red     87899      4   5000.0
2       Toyota   Blue     32549      3   7000.0
3          BMW  Black     11179      5  22000.0
4       Nissan  White    213095      4   3500.0

### 2. Cho biết thuộc tính nào là thuộc tính phân loại, thuộc tính nào là thuộc tính định lượng

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10 entries, 0 to 9
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Manufacturer  10 non-null     object 
 1   Colour        10 non-null     object 
 2   Odometer      10 non-null     int64  
 3   Doors         10 non-null     int64  
 4   Price         10 non-null     float64
dtypes: float64(1), int64(2), object(2)
memory usage: 532.0+ bytes


### 3. Trong dữ liệu có thông tin của bao nhiêu xe?

Đáp án: 10 xe

In [11]:
df["Manufacturer"].count()

10

### 4. Đếm số xe theo theo hãng

Đáp án: Toyota: 10 xe, Honda: 3 xe, Nissan: 2 xe, BMW: 1 xe

In [12]:
df["Manufacturer"].value_counts()

Manufacturer
Toyota    4
Honda     3
Nissan    2
BMW       1
Name: count, dtype: int64

### 5. Đếm số xe theo màu


In [13]:
# Your code here
df["Colour"].value_counts()

Colour
White    4
Blue     3
Red      1
Black    1
Green    1
Name: count, dtype: int64

### 6. Tổng giá trị các xe trong cửa hàng

Đáp án: 76450 USD


In [14]:
# Your code here
df["Price"].sum()

76450.0

### 7. Top 3 xe có giá cao nhất


In [15]:
# Your code here
df.nlargest(3, "Price")

Manufacturer Colour  Odometer  Doors    Price
3          BMW  Black     11179      5  22000.0
9       Nissan  White     31600      4   9700.0
6        Honda   Blue     45698      4   7500.0

### 8. Top 3 xe đã chạy nhiều cây số (km)nhất (Odometer)

Đáp án:
- Xe Nissan, White, 213095 km
- Xe Toyota, White, 150043 km
- Xe Toyota, Green, 99213 km

In [16]:
# Your code here
df.nlargest(3, "Odometer")

Manufacturer Colour  Odometer  Doors   Price
4       Nissan  White    213095      4  3500.0
0       Toyota  White    150043      4  4000.0
5       Toyota  Green     99213      4  4500.0

### 9. Khảo sát xe của hãng Honda

- Tính tỷ lệ xe Honda trong dữ liệu của cửa hàng (0.3)
- Cho biết giá trung bình của xe hãng Honda của cửa hàng (6500)
- Cho biết các xe của hãng Honda trong cửa hàng có những màu sắc nào (Red, Blue)

In [17]:
# Your code here
df[df["Manufacturer"] == "Honda"]["Price"].count() / df["Price"].count()

0.3

In [18]:
df[df["Manufacturer"] == "Honda"]["Price"].mean()

6500.0

In [19]:
tuple(df[df["Manufacturer"] == "Honda"]["Colour"].unique())

('Red', 'Blue')

# 10. Tính toán các giá trị sau của  thuộc tính giá ('Price')

- Giá trung bình các xe (7645)
- Giá trị trung vị các xe (6625)
- Miền giá trị của xe (18500)
- Miền phân vị của xe (2750)
- Điểm phân vị 0.4 và 0.6 của giá ([5750, 7000])

In [20]:
df["Price"].mean()

7645.0

In [21]:
df["Price"].median()

6625.0

In [22]:
df["Price"].max() - df["Price"].min()

18500.0

In [23]:
df["Price"].quantile(0.75) - df["Price"].quantile(0.25)

2750.0

In [24]:
df["Price"].quantile([0.4, 0.6])

0.4    5750.0
0.6    7000.0
Name: Price, dtype: float64

---

# TIPS dataset

Dữ liệu chứa trong file `tips.csv` chứa thông tin tiền tip của khách hàng tại một nhà hàng. Dữ liệu có các thông tin sau:

- `total_bill`: tổng hoá đơn (USD)
- `tip`: tiền tip/ tiền boa (USD)
- `sex`: giới tính người thanh toán hóa đơn
- `smoker`: có hút thuốc hay không
- `day`: ngày trong tuần
- `time`: thời gian dùng bữa(buổi sáng, buổi trưa, buổi tối)
- `size`: số lượng người tham gia

Bạn hãy thực hiện thống kê mô tả để tìm hiểu về hành vi tip của khách hàng tại nhà hàng

1. Đọc dữ liệu


In [25]:
df_tips = load_data("tips")
df_tips.head()

Unnamed: 0  total_bill   tip     sex smoker  day    time  size
0           0       16.99  1.01  Female     No  Sun  Dinner     2
1           1       10.34  1.66    Male     No  Sun  Dinner     3
2           2       21.01  3.50    Male     No  Sun  Dinner     3
3           3       23.68  3.31    Male     No  Sun  Dinner     2
4           4       24.59  3.61  Female     No  Sun  Dinner     4

2. Xoá cột `Unnamed: 0`


In [26]:
df_tips.drop("Unnamed: 0", axis=1, inplace=True)
df_tips.head()

total_bill   tip     sex smoker  day    time  size
0       16.99  1.01  Female     No  Sun  Dinner     2
1       10.34  1.66    Male     No  Sun  Dinner     3
2       21.01  3.50    Male     No  Sun  Dinner     3
3       23.68  3.31    Male     No  Sun  Dinner     2
4       24.59  3.61  Female     No  Sun  Dinner     4

3. Cho biết dữ liệu có bao nhiêu dòng, bao nhiêu thuộc tính.


In [27]:
df_tips.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 244 entries, 0 to 243
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   total_bill  244 non-null    float64
 1   tip         244 non-null    float64
 2   sex         244 non-null    object 
 3   smoker      244 non-null    object 
 4   day         244 non-null    object 
 5   time        244 non-null    object 
 6   size        244 non-null    int64  
dtypes: float64(2), int64(1), object(4)
memory usage: 13.5+ KB


4. Trong các thuộc tính, thuộc tính nào là thuộc tính phân loại, thuộc tính nào là thuộc tính định lượng


In [28]:
df_tips.dtypes.value_counts()

object     4
float64    2
int64      1
Name: count, dtype: int64

5. Có bao nhiêu nam, bao nhiêu nữ. Tỷ lệ bao nhiêu?


In [29]:
df_tips["sex"].value_counts()

sex
Male      157
Female     87
Name: count, dtype: int64

In [30]:
df_tips[df_tips["sex"] == "Male"]["sex"].count() / df_tips[df_tips["sex"] == "Female"][
    "sex"
].count()

1.8045977011494252

6. Có bao nhiêu người hút thuốc. Tỷ lệ bao nhiêu?


In [31]:
df_tips[df_tips["smoker"] == "Yes"]["smoker"].count() / df_tips["smoker"].count()

0.38114754098360654

7. Ngày nào trong tuần khách hàng hay đến dùng bữa nhất. Tỷ lệ bao nhiêu?


In [32]:
df_tips["day"].value_counts().idxmax()

'Sat'

In [33]:
df_tips["day"].value_counts().max() / df_tips["day"].count()

0.35655737704918034

8. Nhà hàng phục vụ bữa ăn nào nhiều nhất (Breakfast, Lunch, Dinner)?


In [34]:
df_tips["time"].value_counts().idxmax()

'Dinner'

9. Nhóm khách hàng dùng bữa tại nhà hàng nhiều nhất là bao nhiêu người?


In [35]:
df_tips["size"].value_counts().idxmax()

2

10. Trung bình trong một bữa ăn nhà hàng phục vụ mấy khách?

In [36]:
df_tips["size"].mean()

2.569672131147541

---